main.cpp

```diff
@@ -13,24 +13,51 @@
 #include <boost/asio.hpp>
 #include <boost/bind.hpp>
 #include "server.hpp"
+#include "nginx-configparser/config_parser.h"
 
 int main(int argc, char* argv[])
 {
   try
   {
     // Check command line arguments.
-    if (argc != 4)
+    if (argc != 2)
     {
-      std::cerr << "Usage: http_server <address> <port> <doc_root>\n";
-      std::cerr << "  For IPv4, try:\n";
-      std::cerr << "    receiver 0.0.0.0 80 .\n";
-      std::cerr << "  For IPv6, try:\n";
-      std::cerr << "    receiver 0::0 80 .\n";
+      std::cerr << "Usage: ./webserver <config_file>\n";
       return 1;
     }
 
+    // Parse the config file.
+    NginxConfigParser config_parser;
+    NginxConfig config;
+    config_parser.Parse(argv[1], &config);
+
+    std::string address = "0.0.0.0";
+    std::string port = "80"; // default port if nothing is set in config file
+    std::string docroot = ".";
+
+    // Look for a server block.
+    for (int i = 0; i < config.statements_.size(); i++)
+    {
+      std::shared_ptr<NginxConfigStatement> statement = config.statements_.at(i);
+      // Look for a listen statement.
+      for (int j = 0; j < statement->tokens_.size(); j++)
+      {
+        if (statement->tokens_.at(j) == "server")
+        {
+          std::vector<std::shared_ptr<NginxConfigStatement>> child = statement->child_block_->statements_;
+          if (child.size() == 1 &&
+            child.at(0)->tokens_.size() == 2 &&
+            child.at(0)->tokens_.at(0) == "listen")
+          {
+            port = child.at(0)->tokens_.at(1);
+            std::cout << "***Listening at port " << port << "***\n";
```

Comment:
I don't think we need the ***'s with a simple output. Perhaps if our output was more complex it'd be nice so that it's easier to see but for now they feel unnecessary to me.